### ISSUE TITLE
Include structured metadata in created PRs

### ISSUE THREAD

**Issue by michaelmior** [scala-steward-org/scala-steward#3465]
I'm looking at building some tooling surrounding the PRs generated by Scala Steward. (For example, automerging PRs based on certain conditions.) For this purpose, it would be helpful to have some structured metadata included in the generated PRs. For example, the identifier and name of the package being updated, the new and old version, and the labels that are already available in a comment.

This data can be parsed out of the current PR text, but it would be nice to have something a bit more stable. [Probot: metadata](https://github.com/probot/metadata) has what seems like a reasonable approach of including JSON data in HTML comments such as the below:

```
<!-- probot = {"json": "here"} -->
```

Something like this would be much less brittle than parsing the current PR text. As a more concrete example, for [this PR](https://github.com/dataunitylab/jsonoid-discovery/pull/47), the metadata might look like this:

```
<!-- scala-steward = {"dependencies": [{"groupId": "com.google.protobuf", "artifactId": "protobuf-java", "oldVersion": "3.25.4", "newVersion": "4.28.1", "labels": ["library-update", "early-semver-major", "semver-spec-major", "commit-count:1"]} -->
```

**Comment by mzuehlke** [comment]
@michaelmior I like the idea and the implementation is just a one liner 😄 
Maybe i would be a good idea to place this into an `update` JSON object, so we could add other fields in the future (like labels, applied scalafix rules, this is not exhaustive).

**Comment by michaelmior** [comment]
@mzuehlke Good point. I actually forgot when I saw how easy the implementation was for the updates that getting the labels was actually the first thing I wanted. (Knowing possible semver compatibility can be a good basis for making automatic merging decisions.) I'll mark the PR as a draft for now and look at implementing some of the changes.

**Comment by mzuehlke** [comment]
Solved in #3466
